imap_processing/glows/l0/glows_l0_data.py

```diff
@@ -0,0 +1,153 @@
+    """Data structure for storing GLOWS histogram packet data.
+    Parameters
+    ----------
+    packet : tuple[list]
+        Histogram packet yielded from space_packet_parser.generate_packets
+
+    Attributes
+    ----------
+    L0_KEYS : tuple[str]
+        Data names from the packet
+    data: dict
+        The data from the packet
+
+    Methods
+    -------
+    process_packet(packet)
+        Process the given packet into the data dictionary
+
+    _convert_histogram_data(binary_hist_data)
+        Process the binary histogram data into 8-bit bins, in a list.
+    """
+
+    L0_KEYS: tuple[str] = (
+        "SHCOARSE",
+        "STARTID",
+        "ENDID",
+        "FLAGS",
+        "SWVER",
+        "SEC",
+        "SUBSEC",
+        "OFFSETSEC",
+        "OFFSETSUBSEC",
+        "GLXSEC",
+        "GLXSUBSEC",
+        "GLXOFFSEC",
+        "GLXOFFSUBSEC",
+        "SPINS",
+        "NBINS",
+        "TEMPAVG",
+        "TEMPVAR",
+        "HVAVG",
+        "HVVAR",
+        "SPAVG",
+        "SPVAR",
+        "ELAVG",
+        "ELVAR",
+        "EVENTS",
+        "HISTOGRAM_DATA",
+    )
+
+    def __repr__(self):
+        return str(self.data)
+
+    def __init__(self, packet):
+        self.data = self.process_packet(packet)
+
+    def process_packet(self, packet) -> dict:
+        """Process the GLOWS histogram packet, which is yielded from space_packet_parser
+
+        Parameters
+        ----------
+        packet
+            Histogram packet from space_packet_parser
+        Returns
+        -------
+        data: dict
+            Data dictionary created from the packet, using L0_KEYS
+        """
+        data = {}
+        for key in self.L0_KEYS:
+            if key != "HISTOGRAM_DATA":
+                data[key] = packet.data[key].derived_value
+            else:
+                data[key] = self._convert_histogram_data(packet.data[key].raw_value)
+            if data[key] is None:
+                raise ValueError(f"Missing data for {key} in {packet.data}")
+        return data
+
+    def _convert_histogram_data(self, binary_hist_data: str) -> list[int]:
+        """Convert the raw histogram data into a list by splitting up the raw binary
+        value into 8-bit segments
+        Parameters
+        ----------
+        binary_hist_data: Raw data read from the packet, in binary format
+
+        Returns
+        -------
+        histograms: list[int]
+            List of binned histogram data
+        """
+        # Convert the histogram data from a large raw string into a list of 8 bit values
+        histograms = []
+        for i in range(8, len(binary_hist_data), 8):
+            histograms.append(int(binary_hist_data[i - 8 : i], 2))
+
+        if len(histograms) != 3599:
+            raise ValueError(
+                f"Histogram packet is lacking bins. Expected a count of 3599, "
+                f"actually received {len(histograms)}"
+            )
+
+        return histograms
+
+
+class GlowsDeL0:
```

Comment:
I always struggle with naming initialisms. Do you want it to be `DE` to signify that they are two words?

I'm not sure you need Glows here because you are in the glows module already and that might save some typing, an alternative is something like this to spell it out more explicitly: `L0DirectEvent`?

Reply:
I went back and forth on the `DE` and ultimately left it like this because the three capitals in a row just looked weird. But that's not a strong preference, and it is two words... 

I do like `L0DirectEvent`, so maybe I will just change it to that. Thanks for the suggestion!